### DISCUSSION TITLE
On the Continuous Growth of Memory Consumption

### DISCUSSION THREAD

**Discussion by TI4NLI4NG** [projectdiscovery/nuclei#4374] [General]
I have encountered a problem while using the Nuclei SDK recently. I am using version 3.0.3 to scan for web vulnerabilities. I noticed that the Nuclei program consumes more and more memory during the scan. This can potentially lead to memory exhaustion when scanning a large number of targets. I would like to know which module is responsible for this excessive memory usage and how I can release memory appropriately during the scanning process.

**Comment by TI4NLI4NG** [comment]
This problem exists when the agent is enabled, and I use the agent locally to direct traffic to the burp suite, causing memory to continue to grow until the task ends. I tested and found that there seems to have been this issue since version 2.9.11, which is evident in version 3.0.3.

**Comment by TI4NLI4NG** [comment]
The new version of rawhttp library used in Nuclei has v0.1.23, which can cause memory leakage when the proxy parameter is set. I rolled back the rawhttp version to v0.1.16 and there was no memory leakage issue. I hope this issue can be resolved in the next version.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
